### ISSUE TITLE
[Accessibility]: Auto-indexing page "enqueue" bar doesn't indicate intent or purpose of any components

### ISSUE THREAD

**Issue by Strum355** [sourcegraph/sourcegraph#35254]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33513

### Problem description

After the heading for the auto-indexing jobs page, the user is presented with a bar from which they can paste a "git rev-like" string with a button for enqueuing an auto-indexing job for that rev-like. 

The screen-reader announces the bar with "git revlike" ("revlike" pronounced more like "rivlike"), followed by the text field announced as "head, insertion at end of text, git revlike, edit text". This doesn't indicate the purpose of the text field, and the purpose of the button is _at a stretch_ somewhat implied.

![image](https://user-images.githubusercontent.com/18282288/167682598-57908c76-4106-4334-abab-3df276fc5040.png)

### Expected behavior

The purpose of the entire bar should be explained outside of a vaguely correlated text element, text field and button.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
@Strum355 Any objections to assigning this to GitStart?

**Comment by Strum355** [comment]
@umpox no objections 👍

**Comment by umpox** [comment]
@Strum355 Just trying to set some AC, I think this benefits everyone on this page! Thoughts on the below wording?

> Provide a [Git revspec](https://git-scm.com/docs/git-rev-parse.html#_specifying_revisions) to enqueue a new auto-indexing job

This is similar copy to the [compare page](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/compare)

**Comment by Strum355** [comment]
> @Strum355 Just trying to set some AC, I think this benefits everyone on this page! Thoughts on the below wording?
> 
> > Provide a [Git revspec](https://git-scm.com/docs/git-rev-parse.html#_specifying_revisions) to enqueue a new auto-indexing job
> 
> This is similar copy to the [compare page](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/compare)

I think that sounds pretty good :+1: @efritz any objections/alterations?

**Comment by efritz** [comment]
SGTM!
